test/runnable/test22342.c

```diff
@@ -0,0 +1,29 @@
+/* RUN_OUTPUT:
+---
+B
+A
+B
+---
+ */
+
+// https://issues.dlang.org/show_bug.cgi?id=22342
+
+int printf(const char *, ...);
+
+int foo()
+{
+  printf("A\n");
+  return 0;
+}
+
+int bar()
+{
+  printf("B\n");
+  return 0;
+}
+
+int main()
+{
+  int v;
+  return bar(1, &v, foo(), "str", bar());
```

Comment:
@WalterBright: Is this really the expected evaluation order - `bar, foo, bar`? I'd have expected `foo, bar, bar`, left-to-right.

Reply:
Usually it depends on what order the target ABI pushes arguments in. Pretty much every single architecture does left-to-right except x86 which is right-to-left.

It is otherwise undefined behaviour in the spec so whatever the order it gets evaluated in, best to ignore it.

Reply:
Thx - according to godbolt and x64 targets, it's left-to-right for clang v13.0.1 and right-to-left for gcc v11.2.

> best to ignore it

So shall we remove the printfs and expected output? [LDC evals left-to-right.]